source/common/router/config_impl.cc

```diff
@@ -404,14 +405,33 @@ void RouteEntryImplBase::finalizeResponseHeaders(
 absl::optional<RouteEntryImplBase::RuntimeData>
 RouteEntryImplBase::loadRuntimeData(const envoy::api::v2::route::RouteMatch& route_match) {
   absl::optional<RuntimeData> runtime;
-  if (route_match.has_runtime()) {
-    RuntimeData data;
-    data.key_ = route_match.runtime().runtime_key();
-    data.default_ = route_match.runtime().default_value();
-    runtime = data;
+  RuntimeData data;
+
+  if (route_match.has_runtime_fraction()) {
+    data.numerator_key_ = route_match.runtime_fraction().numerator_key();
+    data.numerator_default_ = route_match.runtime_fraction().default_value().numerator();
+
+    const std::string& denominator_key = route_match.runtime_fraction().denominator_key();
+    const std::string& denominator_data = loader_.snapshot().get(denominator_key);
+    using FractionalPercent = envoy::type::FractionalPercent;
+    FractionalPercent::DenominatorType denominator_type;
+    if (!FractionalPercent::DenominatorType_Parse(denominator_data, &denominator_type)) {
+      denominator_type = route_match.runtime_fraction().default_value().denominator();
+    }
```

Comment:
I think what you propose is the sane thing to do. I'll change the logic.

Reply:
Actually- scratch that. A user would have unexpected behavior either way if they misspell a denominator type and it switches to a default value. I'm not sure one way is better than the other. @htuch what do you think?

Reply:
Not sure what the best practice to alert when runtime is wrong; how do folks operationally determine when a runtime change has taken effect? Via `/runtime`?

One thought that occurred to me just now is that instead of having separate values for numerator/denominator, we have a single values, which is a textual representation of a `FractionalPercent` proto. That way, we can just use PGV validation on this to check sanity. It also solves an update atomicity issue.